### ISSUE TITLE
Support of (truncated) cones

### ISSUE THREAD

**Issue by nilswagner** [nschloe/pygmsh#10]
https://en.wikipedia.org/wiki/Frustum

**Comment by nschloe** [comment]
If Gmsh support truncated cones, pygmsh will. Since this has been going over a year without upvotes, it's probably popular enough to warrant a separate `add_frustrum()` method.

**Comment by nschloe** [comment]
FYI, pygmsh 4.0 (just released) now supports frustra via Gmsh 3.0's new OpenCASCADE interface:
```python
    geom = pygmsh.opencascade.Geometry(
        characteristic_length_min=0.1,
        characteristic_length_max=0.1,
        )                                              
    geom.add_cone([0.0, 0.0, 0.0], [0.0, 0.0, 1.0], 1.0, 0.3)                                                             
    points, cells, _, _, _ = pygmsh.generate_mesh(geom)
```
